libraries/stdlib/js/src/kotlin/collections/ArrayList.kt

```diff
@@ -71,7 +71,8 @@ public actual open class ArrayList<E> internal constructor(private var array: Ar
         checkIsMutable()
         if (elements.isEmpty()) return false
 
-        array += elements.toTypedArray<Any?>()
+
+        elements.forEach(array.asDynamic()::push)
```

Comment:
Would be better to expand the array by setting its length and then fill elements manually.